### PR TITLE
SDP-2072: Fix payment amount precision loss in dispatch and transaction builder

### PR DESCRIPTION
**Pull request by JiahuiWho**
### What

- Replace `strconv.ParseFloat` + `decimal.NewFromFloat` with `decimal.NewFromString` in `StellarPaymentDispatcher.sendPaymentsToTSS`, eliminating the float64 round-trip that could silently alter payment amounts                             
- Change `StringFixed(6)` to `StringFixed(7)` in `PaymentTransactionHandler.BuildInnerTransaction` to 
  match Stellar's native 7-decimal-place (stroop) precision                                         
- Add precision regression test covering 7dp amounts including boundary values  

### Why

Fix amount precision loss
https://hackerone.com/reports/3683774

### Known limitations

N/A

### Checklist

- [x] Title follows `SDP-1234: Add new feature` or `Chore: Refactor package xyz` format. The Jira ticket code was included if available.
- [x] PR has a focused scope and doesn't mix features with refactoring
- [x] Tests are included (if applicable)
- [x] `CHANGELOG.md` is updated (if applicable)
- [ ] If contracts changed, run the `Contract WASM Artifacts` workflow and open a PR to update the WASMs on `dev`
- [ ] CONFIG/SECRETS changes are updated in helmcharts and deployments (if applicable)
- [ ] Preview deployment works as expected
- [ ] Ready for production

### CODE REVIEW

File: internal/services/paymentdispatchers/stellar_payment_dispatcher.go [hunk 1]
```diff
@@ -59,12 +58,15 @@ var _ PaymentDispatcherInterface = (*StellarPaymentDispatcher)(nil)
...
 	var transactions []txSubStore.Transaction
 	for _, payment := range pendingPayments {
-		// TODO: change TSS to use string amount [SDP-483]
-		amount, err := strconv.ParseFloat(payment.Amount, 64)
+		amount, err := decimal.NewFromString(payment.Amount)
 		if err != nil {
 			return fmt.Errorf("parsing payment amount %s for payment ID %s: %w", payment.Amount, payment.ID, err)
 		}
 
+		if amount.Exponent() < -7 {
```
Comment (marwen-abid) [inline internal/services/paymentdispatchers/stellar_payment_dispatcher.go]: nit:  switch to a value-based check. `amount.Exponent() < -7` is a string-exponent check and would break for padded amounts like `1.00000000`. 

This is a `nit` because currently not an issue due to this `amount` coming from the database, and the db field already normalizes the inputs to 7 fractional digits

```go
const stellarMaxDecimalPlaces = 7
...
if !amount.Equal(amount.Truncate(stellarMaxDecimalPlaces)) {
    return fmt.Errorf("payment amount %s for payment ID %s exceeds Stellar's %d decimal place precision", payment.Amount, payment.ID, stellarMaxDecimalPlaces)
}
```

**Review by marwen-abid — APPROVED** [review]
This is great! Found another issue that I fixed here https://github.com/stellar/stellar-disbursement-platform-backend/pull/1116
